mixins/createGroupRouteMixin.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A function to create a mixin to be used where there is an
+ * optional group param in the route.
+ *
+ * Designed to be used with the GroupSelect component.
+ *
+ * @param key used for the remember key and the route segment
+ * @returns a Vue mixin
+ */
+export default function createGroupRouteMixin(key) {
```

Comment:
At the moment assumes the remember key and the route path are the same, e.g. `communityevents` is the remember key and `/communityevents/` is the path, but could added more args to implement that if needed.